jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -168,7 +173,7 @@ protected void generateAcceptableResponse(Request baseRequest, HttpServletReques
             for (String mimeType : acceptable)
             {
                 generateAcceptableResponse(baseRequest, request, response, code, message, mimeType);
-                if (baseRequest.isHandled())
+                if (response.isCommitted() || baseRequest.getResponse().isWritingOrStreaming())
```

Comment:
I fail to understand this `isWritingOrStreaming()`. Isn't it terribly racy?

Reply:
Hmmm....  The logic here needs to be to keep calling handlers until one handles it.   But we can't use isHandled for this, as it could content with the original dispatched thread.   isCommitted is OK, but not all error pages commit - some just write... but then some might start async processing.

Now I'm not so concerned about racing with isCommitted or writing/streaming, because if the main thread is modifying those, then then application is in a race anyway as it has two threads writing a response - results will always be non deterministic.   The problem is that we are missing if a handler started async processing.... we can't check `isAsyncStarted` because we may already be async????    So perhaps we just don't allow async processing to be done here... after all this is our fail safe just write an acceptable error page NOW method.   So I will say in the javadoc that the method must handle it.

If so, then I think the committed or writing/streaming test is OK.   A boolean return would have been better (as suggested by the OP), but we can't change this API as it is something we encourage users to implement.

Reply:
Updated javadoc... other than that, I think we are good on this part.